src/XMakeBuildEngine/Evaluation/Evaluator.cs

```diff
@@ -2221,26 +2224,36 @@ private List<ProjectRootElement> ExpandAndLoadImports(string directoryOfImportin
                     continue;
                 }
 
-                var newExpandedImportPath = importElement.Project;
-                newExpandedImportPath = newExpandedImportPath.Replace(extensionPropertyRefAsString, extensionPathExpanded);
-
+                var newExpandedImportPath = importElement.Project.Replace(extensionPropertyRefAsString, extensionPathExpanded);
                 _loggingService.LogComment(_buildEventContext, MessageImportance.Low, "TryingExtensionsPath", newExpandedImportPath, extensionPathExpanded);
 
                 List<ProjectRootElement> projects;
-                var result = ExpandAndLoadImportsFromUnescapedImportExpression(directoryOfImportingFile, importElement, newExpandedImportPath,
-                                        false, out projects);
+                var result = ExpandAndLoadImportsFromUnescapedImportExpression(directoryOfImportingFile, importElement, newExpandedImportPath, false, out projects);
 
                 if (result == LoadImportsResult.ProjectsImported)
                 {
-                    return projects;
+                    // If we don't have a wildcard and we had a match, we're done.
+                    if (!containsWildcards)
+                    {
+                        return projects;
+                    }
+
+                    allProjects.AddRange(projects);
                 }
 
                 if (result == LoadImportsResult.FoundFilesToImportButIgnored)
                 {
                     // Circular, Self import cases are usually ignored
                     // Since we have a semi-success here, we stop looking at
                     // other paths
-                    return projects;
+
```

Comment:
Can we add or expand the existing test for this case too? And for the `  ThrowForImportedProjectWithSearchPathsNotFound(fallbackSearchPathMatch, importElement)` code path below. Like maybe a file from a 3rd search path that tries to import itself?